core/src/main/java/org/elasticsearch/common/xcontent/support/XContentMapValues.java

```diff
@@ -418,25 +422,35 @@ public static long nodeLongValue(Object node) {
     /**
      * This method is very lenient, use {@link #nodeBooleanValue} instead.
      */
-    public static boolean lenientNodeBooleanValue(Object node, boolean defaultValue) {
+    public static boolean lenientNodeBooleanValue(Object node, String name, boolean defaultValue) {
         if (node == null) {
             return defaultValue;
         }
-        return lenientNodeBooleanValue(node);
+        return lenientNodeBooleanValue(node, name);
     }
 
     /**
      * This method is very lenient, use {@link #nodeBooleanValue} instead.
      */
-    public static boolean lenientNodeBooleanValue(Object node) {
+    public static boolean lenientNodeBooleanValue(Object node, String name) {
+        boolean interpretedAsLenient = false;
+        boolean booleanValue;
+
         if (node instanceof Boolean) {
-            return (Boolean) node;
+            booleanValue = (Boolean) node;
+        } else if (node instanceof Number) {
+            interpretedAsLenient = true;
+            booleanValue = ((Number) node).intValue() != 0;
+        } else {
+            String value = node.toString();
+            booleanValue = !(value.equals("false") || value.equals("0") || value.equals("off"));
```

Comment:
Actually, I think this is missing parsing `"no"` as false, in which case this would be breaking, perhaps this should use the helpers in `Booleans` instead?